File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0];
        int ans = 0;
        
        for(int i = 0; i < prices.length; i++){
            min = Math.min(min, prices[i]);
            
            if(prices[i] - min > ans){
                ans = prices[i] - min;
            }
        }
        return ans;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxprofit = 0;
        int minPrice = 1e8;
        for(int i = 0; i < prices.size(); i++){
            if(minPrice > prices[i]){
                minPrice = prices[i];
            }
            maxprofit = max(maxprofit, prices[i] - minPrice);  
        }
        return maxprofit;
    }
};